### ISSUE TITLE
Create Stick/Unsticky Containers like in VisualStudio

### ISSUE THREAD

**Issue by jogibear9988** [node-projects/dock-spawn-ts#3]
![image](https://user-images.githubusercontent.com/364896/59008614-51de3880-882b-11e9-86de-9d6a4ed95249.png)

They can be Sticky at left, right, top or bottom (depends on their dock position)

**Comment by NitnerocL** [comment]
Hi! I would love to have this feature too! Have you started working on it?
I might be able to help, but I would need some support to get started

**Comment by jogibear9988** [comment]
I‘ve not looked into it due lack of time.

**Comment by jogibear9988** [comment]
What help do you need?

**Comment by NitnerocL** [comment]
I'm not familiar with the internals of the lib so it will take me some time to get my head around it. If you had some docs / UML diagrams to share it could help, or maybe give me a tour of the project.

**Comment by jogibear9988** [comment]
Not really... I didn't invent it, only converted to typescript and extended it. I'm also always looking on how to do smth....
But I really don't find it to hard.

**Comment by NitnerocL** [comment]
I'll give it a try and let you know if I have specific interrogations then.

Regarding the specs of this feature, should we try to reproduce the behavior of visual studio, with floating tabs above main content, that hide as soon as they lose focus? Or just have a button to open/hide the whole container? I suppose the latter would be easier to implement at first, what do you think?

**Comment by jogibear9988** [comment]
I would think, the floating would be easier. Because the other one, needs to reposition the other containers ;-)
my main point is, keep the elements in DOM, don't remove them. This was the biggest rework I've done in dockspawn, so every element is kept in the dom and only resized (or hidden). Because we have views with 1000s of elments and removing and adding them takes very long

**Comment by evonox** [comment]
I love your library and I use it extensively in my projects. In case of intereset I am willing to help with this feature, too. I am missing it, too.
